config/asound.conf

```diff
@@ -7,43 +7,40 @@ pcm.!default {
     slave.pcm       "ch0"
 }
 
-# Create source 0
-# Use type plug to autonegotiate connection bitrates/mono/stereo.
+# Create source 0 with volume control and channel conversion
 pcm.ch0 {
-    type                     plug
-    slave {
-        pcm {
-            type             dmix
-            ipc_key          2867
-            ipc_perm         0666
-            slave {
-                pcm          "hw:sndrpihifiberry,0"
-                period_time  0
-                period_size  1024
-                buffer_size  8192
-            }
-        }
-    }
+    type            plug
+    slave.pcm       "ch0_softvol"
+    slave.channels  2
+}
 
-    # Lower the output level to avoid clipping on the negative side of the preamp's
-    # U2/U4/U6/U8 op-amps. The audio signal is centered at 2.5 V and the
-    # op-amps cleanly pass down to ~1.4V, so max (2.5 - 1.4) / sqrt(2) = 0.78 Vrms.
-    # Nominal DAC output is ~1.76 Vrms. So 0.78 / 1.76 = 0.44
-    # Note: this is no longer true in Preamp Rev4 and greater, but
-    # we have to leave this here for backwards compatibility.
-    ttable.0.0               0.44
-    ttable.1.1               0.44
+# Lower the output level to avoid clipping on the negative side of the preamp's
+# U2/U4/U6/U8 op-amps. The audio signal is centered at 2.5 V and the
+# op-amps cleanly pass down to ~1.4V, so max (2.5 - 1.4) / sqrt(2) = 0.78 Vrms.
+# Nominal DAC output is ~1.76 Vrms. So 0.78 / 1.76 = 0.44
+# Note: this is no longer true in Preamp Rev4 and greater, but
+# we have to leave this here for backwards compatibility.
 
-    # For now leaving at 44% to maximize SNR before the amplifiers,
-    # and to leave some headroom for per-zone attenuation settings.
-    # If clipping of the amplifiers is to be fully avoided, reduce further:
-    # Max amp power output: 55 W @ 4 Ohms or 30 W @ 8 Ohms.
-    # Max amp Vrms = sqrt(55*4) = 14.8 V = 23.4 dBV
-    # Max amp Vrms = sqrt(30*8) = 15.5 V = 23.8 dBV
-    # Amp gain = 32.8 dBV
-    # Max amp input = 23.8 dBV - 32.8 dBV = -9 dBV = 0.35 Vrms
-    # So to avoid clipping at 0 dB volume, set ttable to:
-    #   0.35 / 1.76 = 0.2
+pcm.ch0_softvol {
+    type            softvol
+    slave.pcm       "ch0_dmix"
+    control.name    "Ch0 Volume"
+    control.card    0
+    max_dB          -3.57  # Limiting volume to 44% of max; 10 * log10(0.44) = -3.57 dB
```

Comment:
This used to be -6.8 dB, is this -3.57 correct?

Reply:
Oh I see, it should be 20*log10() instead of 10*log10() since we are measuring a change in voltage. So 20*log10(0.78/1.76) ~= -7.1 dBV, looks like I used the slightly less precise `0.78/1.7` before to get the -6.8 dB. Either is probably close enough for the average hardware.